src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHost.cs

```diff
@@ -139,7 +139,7 @@ internal async Task RunAsyncCore(CancellationToken cancellationToken, WebAssembl
         manager.SetPlatformRenderMode(RenderMode.InteractiveWebAssembly);
         await manager.RestoreStateAsync(store);
 
-        if (MetadataUpdater.IsSupported)
+        if (MetadataUpdater.IsSupported && Environment.GetEnvironmentVariable("__BLAZOR_WEBASSEMBLY_LEGACY_HOTRELOAD") != "false")
```

Comment:
`GetEnvironmentVariable` returns null if `__BLAZOR_WEBASSEMBLY_LEGACY_HOTRELOAD` is not found. If we keep the current condition, we will treat not-set same as set-true.